lib/Http1Driver.php

```diff
@@ -223,13 +223,14 @@ public function parser(): \Generator {
                 throw new ClientException("Bad Request: multiple host headers", Status::BAD_REQUEST);
             }
 
-            if (!\preg_match("#^(?:https?://)?([^\s:]+)(?::(\d+))?$#i", $headers["host"][0], $matches)) {
+            if (!\preg_match("#^([A-Z\d\.\-]+|\[[\d:]+\])(?::([1-9]\d*))?$#i", $headers["host"][0], $matches)) {
```

Comment:
What about internationalized domain names? Do they have to be ASCII encoded in `host` headers?

Reply:
Yes, host names must be ASCII in host headers.